File: articles/anchoring-bias-economic-judgment/cpp/anchoring_adjustment.cpp
#include <iostream>

double anchored_estimate(double anchor, double true_value, double adjustment_rate) {
    return anchor + adjustment_rate * (true_value - anchor);
}

double anchoring_bias(double anchor, double true_value, double adjustment_rate) {
    return anchored_estimate(anchor, true_value, adjustment_rate) - true_value;
}

int main() {
    std::cout << "Anchored estimate: " << anchored_estimate(85.0, 65.0, 0.55) << std::endl;
    std::cout << "Anchoring bias: " << anchoring_bias(85.0, 65.0, 0.55) << std::endl;
    return 0;
}

File: articles/anchoring-bias-economic-judgment/rust/anchoring_adjustment.rs
fn anchored_estimate(anchor: f64, true_value: f64, adjustment_rate: f64) -> f64 {
    anchor + adjustment_rate * (true_value - anchor)
}

fn anchoring_bias(anchor: f64, true_value: f64, adjustment_rate: f64) -> f64 {
    anchored_estimate(anchor, true_value, adjustment_rate) - true_value
}

fn main() {
    println!("Anchored estimate: {:.2}", anchored_estimate(85.0, 65.0, 0.55));
    println!("Anchoring bias: {:.2}", anchoring_bias(85.0, 65.0, 0.55));
}
